cpp/src/arrow/memory_pool.cc

```diff
@@ -46,10 +46,19 @@
 // See discussion in https://github.com/jemalloc/jemalloc/issues/1621
 
 #ifdef NDEBUG
-const char* je_arrow_malloc_conf = "oversize_threshold:0";
+const char* je_arrow_malloc_conf =
+    ("oversize_threshold:0,"
+     "dirty_decay_ms:1000,"
+     "muzzy_decay_ms:1000,"
+     "background_thread:true");
```

Comment:
It would be nice to add a comment explaining how we came to these settings.

Reply:
done